### PR TITLE
Hits functions

### PR DESCRIPTION
**Pull request by mmkekic**
This PR adds functions for hits list manipulation - merge_NN_hits and  threshold_hits.

### CODE REVIEW

File: invisible_cities/reco/hits_functions.py
Comment (jerenner) [inline invisible_cities/reco/hits_functions.py:34]: Could the second loop through hits be avoided by simultaneously finding the closest Z and the index of this hit, using something like:
```
z_closest, ihit = min((abs(h.Z-nn_h.Z), ihit) for (ihit, h) in enumerate(hits_to_merge))
```

Comment (mmkekic) [reply invisible_cities/reco/hits_functions.py:34]: The problem is that min outputs one element that satisfy the condition and we can have multiple minimal distance hits...

**Review by jerenner — APPROVED** [review]
Functions to:
- merge NN hits into the closest (in z) neighboring non-NN hits
- remove all hits in a list with charge below a given threshold, adding their energies to hits in the same z-slice.

Various tests are included for each function.

File: invisible_cities/reco/hits_functions.py
Comment (jerenner) [reply invisible_cities/reco/hits_functions.py:34]: I see now - `h_closest` ends up being a list of all those hits - ok then, looks good as is.

**Comment by mmkekic** [comment]
Hi all, whats the status of this? @paolafer , @ausonandres  are you also happy with those functions? They always act on hit.E (changing hit.energy) but as @jjgomezcadenas mentioned we dont ever want to redistribute the corrected energies anyway... Comments please

**Comment by ausonandres** [comment]
I've been looking at the functions, especially at the `merge_NN_hits` one, checking with some test events that the energy is conserved before and after the merging, and it is well shared with the closest hits, so I like it.

**Comment by paolafer** [comment]
This PR has been reviewed by two people and approved, can we rebase and merge, please?

**Comment by carmenromo** [comment]
@jerenner can you review the latest commits? I can't merge...

**Review by jerenner — APPROVED** [review]
Function `split_energy` moved - ok

**Comment by carmenromo** [comment]
Before merging this PR I think the failing test should be resolved

**Comment by mmkekic** [comment]
yeah... it is fixed in the PR of blobs redefinition so we can just wait till that one is merged and rebase

**Comment by carmenromo** [comment]
ok!

**Pull request merged**
